modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/ServiceStateManager.java

```diff
@@ -119,22 +118,26 @@ private void asyncStartService(ServiceStateManagementTaskEntity taskEntity,
      * @param id service id.
      * @return task id.
      */
-    public UUID stopService(UUID id) {
-        ServiceStateManagementTaskType taskType = ServiceStateManagementTaskType.STOP;
-        DeployServiceEntity service = getDeployedServiceAndValidateState(id, taskType);
+    public ServiceOrder stopService(UUID id) {
+        ServiceOrderType taskType = ServiceOrderType.SERVICE_STOP;
+        UUID orderId = UUID.randomUUID();
+        DeployServiceEntity service = getDeployedServiceAndValidateState(id, orderId, taskType);
         OrchestratorPlugin plugin = pluginManager.getOrchestratorPlugin(service.getCsp());
-        ServiceStateManageRequest stopRequest = getServiceManagerRequest(service);
-        ServiceStateManagementTaskEntity newTask = createNewManagementTask(id, taskType);
+        ServiceStateManageRequest stopRequest = getServiceManagerRequest(service, orderId);
+        ServiceOrderEntity newTask = createNewManagementTask(id, orderId, taskType);
         taskExecutor.execute(() -> asyncStopService(newTask, plugin, stopRequest, service));
-        return newTask.getTaskId();
+        ServiceOrder serviceOrder = new ServiceOrder();
+        serviceOrder.setServiceId(newTask.getServiceId());
+        serviceOrder.setOrderId(newTask.getOrderId());
+        return serviceOrder;
     }
 
-    private void asyncStopService(ServiceStateManagementTaskEntity taskEntity,
+    private void asyncStopService(ServiceOrderEntity taskEntity,
```

Comment:
please correct the argument names everywhere.. taskEntity is not the correct one in this case.

Reply:
use serviceOrderTaskEntity instead.